### PR TITLE
Integration bibi car

### PR DESCRIPTION
**Pull request by Stefanxdan**
Scuze pentru desciriea lunga:
Fara AUTH
Pe mai multe orase :(

Un asset de la API-ul lor arata asa:
{
            "Id": "5ba20baa-87d5-4e00-94ad-9f7daa6113e4",
            "Brand": "Smart",
            "Color": "Белый",
            "Model": "ForTwo",
            "ModelImageId": "5bf8dd9a-eacd-44a5-89d4-abb500b8fb33",
            "Lat": 44.7379379,
            "Lon": 37.7802353,
            "LicensePlateFormat": "ru"
}
L am transformat in:

"asset:bibicar:729c54eb-f7bc-45ba-a477-5f9014e1c47a": {
                "providerId": "5sde7Bibicar",
                "name": "Smart ForFour Белый",
                "engineType": "unknown", ///////// as fi pus petrol, dar nu stiu daca e benzina sau motorina masina
                "providerType": "bibicar",
                "chargePercentage": 100, ////////// eu i am dat undefined si se pune singur 100
                "type": "cars",
                "priceDetails": "from 4 ₽ / min in motion", //////////// e hardcodat, au lista de preturi foarte complicata pe site
                "coordinate": {
                    "latitude": 48.81984,
                    "longitude": 44.63413
                },
                "validAt": "2021-03-19T22:15:46.413Z",
                "rawId": "729c54eb-f7bc-45ba-a477-5f9014e1c47a",
                "assetId": "asset:bibicar:729c54eb-f7bc-45ba-a477-5f9014e1c47a",
                "imageUrl": "https://static-host.website.com/providers/bibicar/asset-generic.png",
                "currency": "RUB",  ///////// Rubla ruseasca
                "docked": false  ////// am cautat pe site ul lor si nu trebuie sa o lasi intr-un loc special
            },

INSERT INTO `urbanair-integrations`.`region` (`uniqueName`, `radius`, `center`) VALUES ('Volgograd, Rusia', '25', '{\"latitude\": 48.668732, \"longitude\": 44.594683}');
INSERT INTO `urbanair-integrations`.`region` (`uniqueName`, `radius`, `center`) VALUES ('Rostov-on-Don, Rusia', '40', '{\"latitude\": 47.267525, \"longitude\": 39.745279}');
INSERT INTO `urbanair-integrations`.`region` (`uniqueName`, `radius`, `center`) VALUES ('Anapa,  Rusia', '11', '{\"latitude\": 44.944366, \"longitude\": 37.320951}');
INSERT INTO `urbanair-integrations`.`region` (`uniqueName`, `radius`, `center`) VALUES ('Novorossiysk, Rusia', '13', '{\"latitude\": 44.766069, \"longitude\": 37.706000}');
INSERT INTO `urbanair-integrations`.`region` (`uniqueName`, `radius`, `center`) VALUES ('Gelendžik, Rusia', '10', '{\"latitude\": 44.569634, \"longitude\": 38.075053}');

INSERT INTO `urbanair-integrations`.`providers` (`providerId`, `type`, `ctaColor`, `ctaBgColor`, `iosDeeplinks`, `andDeeplinks`, `active`, `supported`, `metadata`, `crawlable`, `rideType`, `name`) VALUES ('5sde7Bibicar', 'bibicar', '#FFFFFF', '#ff7300', '{\"store\": \"https://apps.apple.com/ru/app/bi-bi-car/id1464034736?ls=1\"}', '{\"store\": \"https://play.google.com/store/apps/details?id=ru.bibicar\"}', '1', '1', '{\"assets\":{\"url\":\"https://bibicar.cartrek.online/api/cars\",\"method\":\"GET\",\"headers\":{\"Host\":\"bibicar.cartrek.online\"}}}', '1', 'hail', 'Bi-Bi. car');

### CODE REVIEW

File: src/api/provider/providers/bibicar/BibicarProvider.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        for (const asset of rawData) {
+            const rawId = asset?.Id;
+            const name = asset?.Brand + " " + asset?.Model + " " + asset?.Color;
+
+            const latitude = asset?.Lat;
+            const longitude = asset?.Lon;
+
+
+            const imageUrl = input?.providerGenericAssetImage;
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/bibicar/BibicarProvider.ts:57]: avand in vedere ca sunt mai multe modele de masini, nu putem afisa aceeasi imagine la toate.
Pentru situatiile de genul putem folosi
`const imageUrl = ProviderType.assetModelUrl(providerId, `${asset?.Brand}${asset?.Model}`);`

File: src/api/provider/providers/bibicar/BibicarProvider.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                continue;
+            }
+
+            const assetLocation = { latitude, longitude };
+            
+
+            const newAsset = new Asset(
+                providerId,
+                name,
+                AssetEngineType.value.unknown,
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/bibicar/BibicarProvider.ts:72]: o sa punem `AssetEngineType.value.petrol` indiferent daca masina e diesel sau benzina

File: src/api/provider/providers/bibicar/BibicarProvider.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                providerType,
+                undefined,
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/bibicar/BibicarProvider.ts:74]: ca sa nu iti mai initializeze automat cu 100 la baterie/rezervor, pune 0 in loc de undefined si `''` acolo unde e nevoie de string

**Comment by Stefanxdan** [comment]
Am facut schimbarile, acum response-ul pentru un asset arata asa:
"asset:bibicar:a3ff2ef3-cf9f-4d08-9901-c55a3577a749": {
                "providerId": "5sde7Bibicar",
                "name": "Polo VW ",
                "engineType": "petrol",
                "providerType": "bibicar",
                "charge": "",
                "chargePercentage": 0,
                "type": "cars",
                "priceDetails": "from 4 ₽ / min in motion",
                "coordinate": {
                    "latitude": 48.7636,
                    "longitude": 44.51664
                },
                "validAt": "2021-03-24T08:33:14.637Z",
                "rawId": "a3ff2ef3-cf9f-4d08-9901-c55a3577a749",
                "assetId": "asset:bibicar:a3ff2ef3-cf9f-4d08-9901-c55a3577a749",
                "imageUrl": "https://static-host.website.com/providers/5sde7Bibicar/models/PoloVW.png",
                "currency": "RUB",
                "docked": false
            }

**Pull request merged**